examples/timelockmultisig.sol

```diff
@@ -0,0 +1,250 @@
+            m.canVote = canVote;
+        }
+
+        MembershipChanged(targetMember, canVote);
+
+    }
+
+    function changeMembers(address[] newMembers, bool canVote) {
+        for (uint i = 0; i < newMembers.length; i++) {
+            changeMembership(newMembers[i], canVote, '');
+        }
+    }
+
+    /* Function to create a new proposal */
+    function newProposal(
+        address beneficiary,
+        uint weiAmount,
+        string jobDescription,
+        bytes transactionBytecode
+    )
+        onlyMembers
+        returns (uint proposalID)
+    {
+        proposalID = proposals.length++;
+        Proposal p = proposals[proposalID];
+        p.recipient = beneficiary;
+        p.amount = weiAmount;
+        p.description = jobDescription;
+        p.proposalHash = sha3(beneficiary, weiAmount, transactionBytecode);
+        p.executed = false;
+        ProposalAdded(proposalID, beneficiary, weiAmount, jobDescription);
+        numProposals = proposalID+1;
+        vote(proposalID, true, '');
+
+        return proposalID;
+    }
+
+
+    /* Function to create a new proposal */
+    function newProposalInEther(
+        address beneficiary,
+        uint etherAmount,
+        string jobDescription,
+        bytes transactionBytecode
+    )
+        onlyMembers
+        returns (uint proposalID)
+    {
+        return newProposal(beneficiary, etherAmount * 1 ether, jobDescription, transactionBytecode);
+    }
+
+    /* function to check if a proposal code matches */
+    function checkProposalCode(
+        uint proposalNumber,
+        address beneficiary,
+        uint etherAmount,
+        bytes transactionBytecode
+    )
+        constant
+        returns (bool codeChecksOut)
+    {
+        Proposal p = proposals[proposalNumber];
+        return p.proposalHash == sha3(beneficiary, etherAmount, transactionBytecode);
+    }
+
+    function vote(
+        uint proposalNumber,
+        bool supportsProposal,
+        string justificationText
+    )
+        onlyMembers
+        returns (uint voteID)
+    {
+        Proposal p = proposals[proposalNumber];         // Get the proposal
+        if (p.voted[msg.sender] == true) throw;         // If has already voted, cancel
+        p.voted[msg.sender] = true;                     // Set this voter as having voted
+        if (supportsProposal) {                         // If they support the proposal
+            p.currentResult++;                          // Increase score
+        } else {                                        // If they don't
+            p.currentResult--;                          // Decrease the score
+        }
+        // Create a log of this event
+        Voted(proposalNumber,  supportsProposal, msg.sender, justificationText);
+    }
+
+    function proposalDeadline(uint proposalNumber) constant returns(uint deadline) {
+        Proposal p = proposals[proposalNumber];
+        uint factor = 3*10**uint(6 - (5 * p.currentResult / int(members.length - 1)));
```

Comment:
p.currentResult should be negative, that is not a problem. If it's negative then the transaction will have to wait an order of magnitude of decades before being executed - but line 233 also prevents any negative anyway.

Adding members would not make it negative, thou, as the members.length can only grow. But you have a point:

If a member is removed (it's not actually removed, it just stays there with the right to vote off), then their vote still count. We could change the code, but then it would run with limits of running trough array lengths..